packages/ui-components/src/Tooltip/Tooltip.tsx

```diff
@@ -67,36 +70,48 @@ export const Tooltip: FunctionComponent<TooltipProps> = ({
       ) : (
         (child as ReactElement)
       ),
-      {
-        ref: setReferenceElement,
-        onMouseOver: () => {
-          popperElement.setAttribute("data-show", "");
-        },
-        onMouseOut: () => {
-          if (visible) return;
-          popperElement.removeAttribute("data-show");
-        },
-      },
     );
   });
 
+  const timers: any = [];
   return (
-    <>
+    <div
+      ref={setReferenceElement}
+      onMouseOver={() => {
+        const t = setTimeout(() => {
+          popperElement.setAttribute("data-show", "");
+        }, timeout);
+        timers.push(t);
+      }}
+      onMouseLeave={() => {
+        timers.forEach((t: any) => {
+          clearTimeout(t);
+        });
+      }}
+    >
       {wrappedChildren}
 
       <div
         className={classnames}
         ref={setPopperElement}
         style={styles.popper}
         data-jest="tooltip"
+        onMouseLeave={e => {
```

Comment:
`e` isn't used. I generally like to have it here to demonstrate that this is a callback that takes an _event_, but this will throw an linter warning so we might want to change it to
```suggestion
        onMouseLeave={() => {
```